watcher.go

```diff
@@ -0,0 +1,107 @@
+package frankenphp
+
+import (
+	fswatch "github.com/dunglas/go-fswatch"
+	"go.uber.org/zap"
+	"sync/atomic"
+	"sync"
+)
+
+// latency of the watcher in milliseconds
+const watcherLatency = 150
+
+var (
+	watchSessions      []*fswatch.Session
+	blockReloading     atomic.Bool
+	reloadWaitGroup    sync.WaitGroup
+)
+
+func initWatcher(watchOpts []watchOpt, workerOpts []workerOpt) error {
+	if(len(watchOpts) == 0 || len(workerOpts) == 0) {
+		return nil
+	}
+
+	watchSessions := make([]*fswatch.Session, len(watchOpts))
+	for i, watchOpt := range watchOpts {
+		session, err := createSession(watchOpt, workerOpts)
+		if(err != nil) {
+			return err
+		}
+		watchSessions[i] = session
+	}
+
+	for _, session := range watchSessions {
+		go session.Start()
+	}
+
+	blockReloading.Store(false)
+	reloadWaitGroup = sync.WaitGroup{}
+	return nil;
+}
+
+func createSession(watchOpt watchOpt, workerOpts []workerOpt) (*fswatch.Session, error) {
+	eventTypeFilters := []fswatch.EventType{
+		fswatch.Created,
+		fswatch.Updated,
+		fswatch.Renamed,
+		fswatch.Removed,
+	}
+	// Todo: allow more fine grained control over the options
+	opts := []fswatch.Option{
+		fswatch.WithRecursive(watchOpt.isRecursive),
+		fswatch.WithFollowSymlinks(false),
+		fswatch.WithEventTypeFilters(eventTypeFilters),
+		fswatch.WithLatency(watcherLatency / 1000),
```

Comment:
nit: we could store the final value directly in the constant.

Reply:
Done 👍